custom_components/sun2/binary_sensor.py

```diff
@@ -118,8 +119,12 @@ def _val_cfg(config):
 class Sun2ElevationSensor(Sun2SensorBase, BinarySensorEntity):
     """Sun2 Elevation Sensor."""
 
-    def __init__(self, hass, name, above, info):
+    def __init__(self, hass, ns, name, above, info):
         """Initialize sensor."""
+        if ns:
+            self._attr_unique_id = f"{ns} {name}"
+        else:
+            self._attr_unique_id = name
```

Comment:
I'm not so sure this is a good idea, this would give sensors like "sensor.daylight" as a unique_id right?

Maybe force a prefix when not using a namespace? or no unique_id if you're not using a ns? 
not sure what the right approach is.

Reply:
Placed this at the wrong method, should have been a comment on the abstract method of sensor.py

Reply:
If namespace was not specified, then the unique ID (in your example) would be "Daylight". There really shouldn't be more than one daylight sensor configured with no namespace (or with a given namespace, for that matter), so that shouldn't be a problem.

The whole idea of using `entity_namespace` is to allow having more than one of a given type of sensor (usually with different location information), such that the resulting entity ID is predictable. (If namespace wasn't used, two daylight sensors, say with two different sets of location values, would have unpredictable entity IDs. E.g., one would be `sensor.daylight` and the other would be `sensor.daylight_2`, and which is which can't be predetermined. At least, that's what I would expect to happen. I haven't tried it recently.)

The actual value of the unique ID isn't important, just that it's unique for all sensors of the same domain (i.e., `sensor` or `binary_sensor`) and for the same platform (i.e., `sun2`.) The unique ID isn't even displayed to the user.

But what is _really_ important is coming up with a method to determine a unique ID based on the config file, such that it _doesn't change_ when HA restarts, and is very unlikely to change if the configuration is changed. This is important because it "maps" the config file entry to the entity registry entry. As long as the `entity_namespace` value isn't changed, the mapping will stay the same.